### PR TITLE
Upgrade Node Engine From 16 to 18

### PR DESCRIPTION
**Pull request by phoenix-clauster**
#### What does this PR do?
Upgrades Node Engine from 16 to 18.

#### Where should the reviewer start?
at package.json.

#### What testing has been done on this PR?
Project was started using npm run storybook to ensure functionality.

#### How should this be manually tested?
The reviewer can run npm run storybook to validate it.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
No

#### What are the relevant issues?
Closes https://github.com/grommet/grommet/issues/6911

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
backwards compatible.

### CODE REVIEW

**Comment by phoenix-clauster** [comment]
Hi @jcfilben,
I see that netlify pipelines are failing due to a node cached version (current 16) and it is requiring version 18.
Does this upgrade requires an update on netlify side?

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
   },
   "engine-strict": true,
   "engines": {
-    "node": ">= 16"
+    "node": "18"
```
Comment (jcfilben) [inline package.json:28]:
```suggestion
    "node": ">= 18"
```
We should allow any version of node that is 18 or above instead of just 18

**Comment by jcfilben** [comment]
> Hi @jcfilben,
I see that netlify pipelines are failing due to a node cached version (current 16) and it is requiring version 18.
Does this upgrade requires an update on netlify side?

I just went in and upgraded the version of node used on our netlify to v18. Next time the checks run the netlify deploy preview should work.

File: package.json (hunk 1 above)
Comment (phoenix-clauster) [reply package.json]: Noted. Thanks!

**Comment by phoenix-clauster** [comment]
Thanks for your feedback. PR has been updated.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by britt6612 — APPROVED** [review]

**Comment by MikeKingdom** [comment]
Didn't we decide we can't do this yet due to backward compatibility? Some products are still on 16 for a bit

**Comment by britt6612** [comment]
> Drop support for node v16 in September #6911

If we did we should add a note to the issue.

**Comment by jcfilben** [comment]
Discussed with the team, we are going to move forward with this. Products that are still using node 16 will just get a warning when they install dependencies.

**Pull request merged**